src/randomizer.py

```diff
@@ -2080,6 +2099,99 @@ def visit_Decl(self, decl: ca.Decl) -> None:
     ast_util.insert_decl(fn, var, type, random)
 
 
+def perm_inline_get_structmember(
+    fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
+) -> None:
+    """Creates an inline function for accessing a struct member."""
+
+    typemap = build_typemap(ast)
+
+    cands: List[Union[ca.StructRef, ca.UnaryOp]] = []
+    for expr in get_block_expressions(fn.body, region):
+        if isinstance(expr, ca.StructRef):
+            cands.append(expr)
+        if (
+            isinstance(expr, ca.UnaryOp)
+            and expr.op is "&"
+            and isinstance(expr.expr, ca.StructRef)
+        ):
+            cands.append(expr)
+
+    ensure(cands)
+
+    cand = random.choice(cands)
+
+    ensure(cands)
+    cand = random.choice(cands)
+
+    parentOp: Optional[ca.UnaryOp] = None
+    if isinstance(cand, ca.UnaryOp):
+        parentOp = cand
+        assert isinstance(parentOp.expr, ca.StructRef)
+        cand = parentOp.expr
+
+    member_type: SimpleType = decayed_expr_type(cand, typemap)
+    member_name = cand.field.name
+
+    new_inline_fn_name = get_noncolliding_name(ast, f"get_{member_name}")
+
+    # Replace the StructRef with a FuncCall to the new inline yet to be created
+    replace_node(
+        fn.body,
+        cand if not parentOp else parentOp,
+        ca.FuncCall(ca.ID(new_inline_fn_name), ca.ExprList([cand.name])),
+    )
+
+    # Now create the new inline function definition
+    arg_type: SimpleType = decayed_expr_type(cand.name, typemap)
+    assert isinstance(arg_type.type, ca.TypeDecl)
+    assert arg_type.type.declname
+
+    theParam = ca.Decl(
+        name=None,
+        quals=[],
+        align=[],
+        storage=[],
+        funcspec=[],
+        type=arg_type,
+        init=None,
+        bitsize=None,
+    )
+
+    fn_decl = ca.Decl(
+        name=new_inline_fn_name,
+        quals=[],
+        align=[],
+        storage=[],
+        funcspec=["inline"],
+        type=ca.FuncDecl(
+            args=ca.ParamList([theParam]),
+            type=copy.deepcopy(member_type),
+        ),
+        init=None,
+        bitsize=None,
+    )
+
+    set_decl_name(fn_decl)
+
+    return_expr: Expression = ca.StructRef(
+        name=ca.ID(arg_type.type.declname),
+        type="->",
```

Comment:
```suggestion
        type=cand.type,
```
or forbid `cand.type == "."` early on

Reply:
thought I had something, that prevented "."   but I guess not and it's added explicitly now